server/etcdserver/raft_test.go

```diff
@@ -171,6 +171,7 @@ func TestStopRaftWhenWaitingForApplyDone(t *testing.T) {
 	select {
 	case <-srv.r.applyc:
 	case <-time.After(time.Second):
+		srv.r.stop()
```

Comment:
The `srv.r.stop()` may be blocked. I would suggest to follow the same way as line 178 ~ 183, which can be wrapped in a function.

Reply:
Done.